atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/asts/bucketingthings/DefaultSweepAssignedBucketStore.java

```diff
@@ -81,6 +97,83 @@ public BucketStateAndIdentifier getBucketStateAndIdentifier() {
                         + " initialisation, and as such, is an invalid state."));
     }
 
+    @Override
+    public Set<Bucket> getStartingBucketsForShards(Set<ShardAndStrategy> shardAndStrategies) {
+        Map<ShardAndStrategy, Cell> shardAndStrategyToCell = shardAndStrategies.stream()
+                .collect(Collectors.toMap(
+                        Function.identity(), SweepAssignedBucketStoreKeyPersister.INSTANCE::sweepBucketPointerCell));
+        Map<Cell, Long> results = readCells(shardAndStrategyToCell.values(), bucketIdentifierPersister::tryDeserialize);
+        return KeyedStream.stream(shardAndStrategyToCell)
+                .map(cell -> Optional.ofNullable(results.get(cell)))
+                .map((shardAndStrategy, bucketIdentifier) -> bucketIdentifier
+                        .map(id -> Bucket.of(shardAndStrategy, id))
+                        .orElseGet(() -> {
+                            log.warn(
+                                    "No starting bucket found for shard and strategy {}",
```

Comment:
We should probably say what we're doing (starting from 0), and also that this is expected when starting with autoscaling targeted sweep for the first time? Wouldn't want operators to get too alarmed if they see a pile of these on startup